### PR TITLE
UI components Update: Tabs, Icons, Iconbuttons

### PR DESCRIPTION
**Pull request by niklasdfx**
Added 

* TabComponent
* Icons
* IconButtons

Updated

* Databox: can have integrated button ("log out")
  * can have settings

### CODE REVIEW

File: src/stories/StyledDataBox.tsx [hunk 1]
```diff
@@ -1,14 +1,59 @@
 import { PropsWithChildren } from 'react';
+import { IconColors, IconSizes, IconVariant } from './DfxIcon';
+import StyledIconButton from './StyledIconButton';
 
-interface IStyledDataBoxProps extends PropsWithChildren {
+interface StyledDataBoxProps extends PropsWithChildren {
   heading: string;
+  loggedIn?: boolean;
+  hasSettings?: boolean;
```
[the remark this answers is not in the document]
Comment (davidleomay) [reply src/stories/StyledDataBox.tsx]: Yes I agree, we definitely should find generic inputs for it.

File: src/stories/StyledModal.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            <div className="relative p-6 flex-auto">
+              <p className="my-4 text-slate-500 text-lg leading-relaxed">
+                I always felt like I could do anything. That’s the main thing people are controlled by! Thoughts- their
+                perception of themselves! They're slowed down by their perception of themselves. If you're taught you
+                can’t do anything, you won’t do anything. I was taught I could do everything.
+              </p>
+            </div>
+            {/*footer*/}
+            <div className="flex items-center justify-end p-6 border-t border-solid border-slate-200 rounded-b">
+              <StyledButton label="Close Modal" onClick={() => setShowModal(false)}></StyledButton>
```
Comment (davidleomay) [inline src/stories/StyledModal.tsx:49]: `label="Close"`?

File: src/stories/StyledModal.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import StyledButton from './StyledButton';
+
+export enum StyledModalColors {
+  GRAY = 'GRAY',
+  WHITE = 'WHITE',
+  DFXGRADIENT = 'DFXGRADIENT',
+}
+
+interface StyledModalProps extends PropsWithChildren {
+  setShowModalFunc: (showModal: boolean) => any;
```
Comment (davidleomay) [inline src/stories/StyledModal.tsx:11]: An `onClose` prop would be simpler 😁

File: src/stories/StyledTab.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  tabId,
+  children,
+  type,
+  openTab,
+  updateActiveTab,
+  deactivated = false,
+}: StyledTabProps) {
+  const hrefLink = '#link' + tabId;
+  let tabClasses = 'text-2xl font-black px-12 py-2 rounded-t-lg block flex gap-2 ';
+  if (deactivated === false) {
```
Comment (davidleomay) [inline src/stories/StyledTab.tsx:32]: `if (!deactivated) {`

File: src/stories/StyledDataBox.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (niklasdfx) [reply src/stories/StyledDataBox.tsx]: thanks for the suggestion. i replaced hassetting with rightIconButton and loggedIn with a  (if boxButtonLabel !== undefined)-logic

File: src/stories/StyledDataBox.tsx [hunk 5]
```diff
@@ -1,14 +1,59 @@
...
+    <div className="mb-6 sm:max-w-lg md:max-w-none md:w-full mx-auto">
+      <div className={containerClasses}>
+        <div className="flex justify-between content-start">
+          <h2 className={headingClasses}>{heading}</h2>
+          {hasSettings && (
+            <StyledIconButton
+              icon={IconVariant.SETTINGS}
+              color={IconColors.RED}
+              size={IconSizes.LG}
+              onClick={() => {
```
[the remark this answers is not in the document]
Comment (niklasdfx) [reply src/stories/StyledDataBox.tsx]: thanks for the suggestion. i replaced hassetting with rightIconButton and loggedIn with a (if boxButtonLabel !== undefined)-logic

File: src/stories/StyledTab.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
+import { PropsWithChildren, ReactElement } from 'react';
+import DfxIcon, { IconColors, IconSizes, IconVariant } from './DfxIcon';
+import { TabType } from './StyledTabContainer';
+
+interface StyledTabProps extends PropsWithChildren {
+  tabId: number;
+  updateActiveTab: (tId: number) => any;
+  openTab: number;
```
Comment (davidleomay) [inline src/stories/StyledTab.tsx:17]: Consistent naming would make it easier to use, e.g. `activeTab` and `setActiveTab`

File: src/stories/StyledDataBox.tsx [hunk 7]
```diff
@@ -1,14 +1,59 @@
...
+                console.log('clicked');
+              }}
+            />
+          )}
+        </div>
+        <div>{children}</div>
+      </div>
+      {loggedIn && (
```
[the remark this answers is not in the document]
Comment (niklasdfx) [reply src/stories/StyledDataBox.tsx]: went with ` if boxButtonLabel !== undefined`

File: src/stories/StyledTab.tsx (hunk 6 above)
Comment (davidleomay) [reply src/stories/StyledTab.tsx:17]: Even simpler would be just `active` and `setActive`? The tab probably doesn't care about its ID 😉

File: src/stories/StyledTabContainer.tsx [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  type?: TabType;
+};
+
+export enum TabType {
+  BUY = 'BUY',
+  SOON = 'COMING SOON',
+}
+
+export default function StyledTabContainer({ tabs, openTabOnLoad = 0 }: StyledTabContainerProps) {
+  const [openTab, setOpenTab] = React.useState(openTabOnLoad);
```
Comment (davidleomay) [inline src/stories/StyledTabContainer.tsx:17]: `React` can be omitted:
`const [openTab, setOpenTab] = useState(openTabOnLoad);`

File: src/stories/StyledTab.tsx [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+  deactivated?: boolean;
+  type?: TabType;
+}
+
+const TABTYPE_MAP: Record<TabType, ReactElement> = {
+  [TabType.BUY]: <IconTab icon={IconVariant.BANK} />,
+  [TabType.SOON]: <ComingSoon />,
```
Comment (davidleomay) [inline src/stories/StyledTab.tsx]: I think this should be generic and be exposed as parameter. Maybe export the `IconTab` component?

File: src/stories/StyledModal.tsx (hunk 2 above)
Comment (niklasdfx) [reply src/stories/StyledModal.tsx:49]: sorry, this is just work-in-progress-content. i was working on modal and realized i need a icon-button for it, so i created icons and iconbuttons first ... so this "close"-button will not be there anyway :)

File: src/stories/StyledModal.tsx (hunk 3 above)
Comment (niklasdfx) [reply src/stories/StyledModal.tsx:11]: you're right :)

File: src/stories/StyledTab.tsx [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+  return (
+    <li className="flex-none text-center ">
+      <a
+        className={tabClasses}
+        onClick={(e) => {
+          e.preventDefault();
+          if (deactivated === false) {
+            setCurrentTab();
+          }
```
[the remark this answers is not in the document]
Comment (niklasdfx) [reply src/stories/StyledTab.tsx:50]: thanks, i was wondering, if there is a shorter way :)

File: src/stories/StyledTab.tsx (hunk 9 above)
Comment (niklasdfx) [reply src/stories/StyledTab.tsx]: did a re-write of this functionality. now the icon or flagwords can be passed down and the visual-addon is generated according to the case ...

File: src/stories/StyledTab.tsx (hunk 6 above)
Comment (niklasdfx) [reply src/stories/StyledTab.tsx]: changed it.

File: src/stories/StyledTabContainer.tsx (hunk 8 above)
Comment (niklasdfx) [reply src/stories/StyledTabContainer.tsx]: changed.

File: src/stories/StyledModal.tsx [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+
+export enum StyledModalColors {
+  GRAY = 'GRAY',
+  WHITE = 'WHITE',
+  DFXGRADIENT = 'DFXGRADIENT',
+}
+
+interface StyledModalProps extends PropsWithChildren {
+  onClose: (showModal: boolean) => any;
+  closeWithX?: boolean;
```
Comment (davidleomay) [inline src/stories/StyledModal.tsx:12]: `closeWithX` is unused

File: src/stories/DfxIcon.tsx [hunk 12]
```diff
@@ -101,7 +101,11 @@ const SIZE_MAPS: Record<IconSizes, string> = {
   [IconSizes.XL]: '32px',
 };
 
-export default function DfxIcon({ icon, color = IconColors.RED, size = IconSizes.MD }: DfxIconProps) {
+export default function DfxIcon({
+  icon = IconVariant.BANK,
```
[the remark this answers is not in the document]
Comment (niklasdfx) [reply src/stories/DfxIcon.tsx:105]: no, you're right :)

File: src/stories/StyledModal.tsx (hunk 11 above)
Comment (niklasdfx) [reply src/stories/StyledModal.tsx:12]: yeah, there's more work on modal ahead ... it will be used :)

Comment (niklasdfx) [reply src/stories/StyledModal.tsx:12]: needed to build the iconButtons first

**Pull request merged**